hw4/04-templates.js

```diff
@@ -12,6 +12,54 @@ app.set('view engine', 'pug');
 const url = 'https://restcountries.com/v3.1/all';
 
 // Add your code here
+const data = [];
+const getData = (url) => {
+  axios
+    .get(url)
+    .then((response) => {
+      response.data.forEach((country) => {
+        data.push(country);
+      });
+    })
+    .catch(() => {
+      console.log('An error occured. Please reload the page.');
+    });
+};
+
+getData(url);
+
+const getList = (route) => {
+  const list = [];
+  if (route === '/capitals') {
+    data.sort((a, b) => a.name.common.localeCompare(b.name.common));
+    data.forEach((country) => {
+      const capital = country.capital || 'no data';
+      list.push(`${country.name.common} - ${capital}`);
+    });
+  }
+  if (route === '/populous') {
+    data.sort((a, b) => b.population - a.population);
+    data.forEach((country) => {
+      if (country.population >= 50000000) {
+        list.push(`${country.name.common} - ${country.population}`);
+      }
+    });
+  }
+  if (route === '/regions') {
+    const map = new Map();
```

Comment:
I didn't know about Map functionality in JS until I saw this. Amazing idea to solve your solution this way!